packages/web-pkg/src/components/sidebar/SideBar.vue

```diff
@@ -0,0 +1,336 @@
+              class="sidebar-panel__navigation"
+            >
+              <oc-button
+                v-for="panelSelect in availablePanels.filter((p) => !p.default)"
+                :id="`sidebar-panel-${panelSelect.app}-select`"
+                :key="`panel-select-${panelSelect.app}`"
+                :data-testid="`sidebar-panel-${panelSelect.app}-select`"
+                appearance="raw"
+                @click="openPanel(panelSelect.app)"
+              >
+                <oc-icon :name="panelSelect.icon" :fill-type="panelSelect.iconFillType" />
+                {{ panelSelect.component.title($gettext) }}
+                <oc-icon name="arrow-right-s" fill-type="line" />
+              </oc-button>
+            </div>
+          </template>
+          <p v-else>{{ sidebarAccordionsWarningMessage }}</p>
+        </div>
+      </div>
+    </template>
+  </div>
+</template>
+
+<script>
+import { mapState, mapActions } from 'vuex'
+import { VisibilityObserver } from 'web-pkg/src/observer'
+
+let visibilityObserver
+let hiddenObserver
+
+export default {
+  props: {
+    loading: {
+      type: Boolean,
+      required: true
+    },
+    availablePanels: {
+      type: Array,
+      required: true
+    },
+    sidebarAccordionsWarningMessage: {
+      type: String,
+      required: false,
+    },
+    isContentDisplayed: {
+      type: Boolean,
+      required: false,
+      default: true
+    },
+    // FIXME: remove
+    isSingleResource: {
+      type: Boolean,
+      required: true,
+    }
+  },
+
+  data() {
+    return {
+      focused: undefined,
+      oldPanelName: null,
+      selectedFile: {},
+    }
+  },
+
+  computed: {
+    ...mapState('Files/sidebar', { sidebarActivePanel: 'activePanel' }),
+    activeAvailablePanelName() {
+      if (!this.sidebarActivePanel) {
+        return null
+      }
+      if (!this.availablePanels.map((p) => p.app).includes(this.sidebarActivePanel)) {
+        return null
+      }
+      return this.sidebarActivePanel
+    },
+    activePanelName() {
+      return this.activeAvailablePanelName || this.defaultPanel.app
+    },
+    defaultPanel() {
+      return this.availablePanels.find((panel) => panel.default)
+    },
+    accessibleLabelBack() {
+      const translated = this.$gettext('Back to %{panel} panel')
+      return this.$gettextInterpolate(translated, {
+        panel: this.defaultPanel.component.title(this.$gettext)
+      })
+    },
+  },
+  watch: {
+    activePanelName: {
+      handler: function (panel, select) {
+        this.$nextTick(() => {
+          this.focused = panel ? `#sidebar-panel-${panel}` : `#sidebar-panel-select-${select}`
+        })
+      },
+      immediate: true
+    },
+  },
+
+  beforeDestroy() {
+    visibilityObserver.disconnect()
+    hiddenObserver.disconnect()
+  },
+  methods: {
+    ...mapActions('Files/sidebar', {
+      closeSidebar: 'close',
+      setSidebarPanel: 'setActivePanel',
+      resetSidebarPanel: 'resetActivePanel'
+    }),
```

Comment:
We should not track the state in the files app vuex namespace.

Should we use the store at all? Personally I'd like to store the state in the url... but I see that not everyone might like that behavior (sidebar stays open after page reload)